packages/tic-tac-toe/app/controllers/game.ts

```diff
@@ -28,19 +49,119 @@ export default class GameController extends Controller {
     return arr;
   }
 
+  @action
+  protected async initGame(): Promise<void> {
+    if (this.currentGame.getPlayer() === Player.B) {
+      console.log('Starting game as Player B. Waiting for Player A to play.');
+    } else {
+      console.log('Starting game as Player A. My turn. Need to play');
+    }
+
+    const updatesId = Date.now();
+    this.channelUpdates.subscribeToMessages(
+      updatesId,
+      (channelState: ChannelState.ChannelState<AppData>) => {
+        if (this.currentGame.getPlayer() === Player.B) {
+          if (ChannelState.isRunning(channelState)) {
+            if (ChannelState.inXPlaying(channelState)) {
+              this.player = 'o';
+              this.Xs = channelState.appData.Xs;
+              this.Os = channelState.appData.Os;
+              console.log('Moving from xPlaying -> oPlaying');
+            }
+            if (ChannelState.inOPlaying(channelState)) {
+              this.player = 'x';
+              this.Xs = channelState.appData.Xs;
+              this.Os = channelState.appData.Os;
+              console.log('Moving from oPlaying -> xPlaying');
+            }
+          }
+        } else {
+          if (ChannelState.isRunning(channelState)) {
+            if (ChannelState.inXPlaying(channelState)) {
+              this.player = 'o';
+              this.Xs = channelState.appData.Xs;
+              this.Os = channelState.appData.Os;
+              console.log('Moving from xPlaying -> oPlaying');
+            }
+            if (ChannelState.inOPlaying(channelState)) {
+              this.player = 'x';
+              this.Xs = channelState.appData.Xs;
+              this.Os = channelState.appData.Os;
+              console.log('Moving from oPlaying -> xPlaying');
+            }
+          }
```

Comment:
same than my comment below. This looks, sort of, like repeated code

Reply:
Done thanks